cmake/modules/RootBuildOptions.cmake

```diff
@@ -301,6 +305,15 @@ endif()
 #---Define at moment the options with the selected default values-----------------------------
 ROOT_APPLY_OPTIONS()
 
+if(root7)
+  if(NOT CMAKE_CXX_STANDARD GREATER 11)
+      set(CMAKE_CXX_STANDARD 14 CACHE STRING "C++14 standard used with root7")
```

Comment:
I think changing the `CMAKE_CXX_STANDARD` without checking a user-provided value is bad. Probably even with checking a user-provided value. Can we error out saying "please set -DCMAKE_CXX_STANDARD=14 or higher"?

Reply:
>  Can we error out saying "please set -DCMAKE_CXX_STANDARD=14 or higher"?

Probably my code is imperfect, but we should handle situation when only `-Droot7=ON` is specified. I guess, it is no problem to force c++14 in such case. Only when `-DCMAKE_CXX_STANDARD=11` configured by user, one should abort with error

Reply:
That is default value (if any) for `CMAKE_CXX_STANDARD` variable?

Reply:
>  Only when -DCMAKE_CXX_STANDARD=11 configured by user, one should abort with error

That's fine with me. Can you look into what the default value of `CMAKE_CXX_STANDARD` is? (Or whether ROOT's CMake files have set it already when getting to this point?)

Reply:
> Can you look into what the default value of CMAKE_CXX_STANDARD is?

I found very small info about it. This is just default value for `CXX_STANDARD`. In my environment (cmake 3.13.1, OpenSuSE Linux) it is undefined at this place.

Reply:
Thanks, then let's just error out if `CMAKE_CXX_STANDARD` is defined, else set it (with a `message`, please). Could you update the PR accordingly?